src/Database/Type/IntegerType.php

```diff
@@ -98,7 +98,7 @@ public function toPHP($value, Driver $driver)
     public function manyToPHP(array $values, array $fields, Driver $driver)
     {
         foreach ($fields as $field) {
-            if (!isset($values[$field])) {
+            if (!isset($values[$field]) || !is_numeric($values[$field])) {
```

Comment:
It should throw instead of drop the value

Reply:
:+1: we drop values in marshalling, and raise when converting to PHP or DB values.